addons/crm/models/crm_lead.py

```diff
@@ -1749,34 +1713,60 @@ def _handle_salesmen_assignment(self, user_ids=False, team_id=False):
     # CLASSIFICATION TOOLS
     # --------------------------------------------------
 
-    def _get_lead_duplicates(self, partner=None, email=None, include_lost=False):
-        """ Search for leads that seem duplicated based on partner / email.
+    def _get_lead_duplicates(self, partner=None, email=None, phone_sanitized=None, include_lost=False):
+        """ Search for leads that seem duplicated based on partner / email / phone.
 
         :param partner : optional customer when searching duplicated
-        :param email: email (possibly formatted) to search
-        :param boolean include_lost: if True, search includes archived opportunities
-          (still only active leads are considered). If False, search for active
-          and not won leads and opportunities;
+        :param email: email domain exact match
```

Comment:
Weird docstring ? Isn't it like "email used to find matches based on either full email (using providers like gmail) or domain part (using company email domains)" ?

Reply:
Now, we compare only the domain part here.
isn't using email domain sufficient? rather than adding `or` with full email

Reply:
If you look at "mail_prepare_for_domain_search", it is either a full email, either a domain. That's the purpose of "email criterion"